frontend/src/components/organisms/MatchesList/MatchesList.tsx

```diff
@@ -23,144 +24,227 @@ export type MatchesListProps = {
    return (
       <StyledMatchesList>
-         {matchesListItems ? matchesListItems.map((item, index) => returnMatchItem(item, index, expanded, handleChangeExpander)) : "Oczekiwanie na pobranie..."}
+         {matchesList
+            ? matchesList
+               .map((item, index) => returnMatchItem(item, index, expanded, handleChangeExpander))
+            : "Oczekiwanie na pobranie..."}
       </StyledMatchesList>
    )
 };
 
-const returnMatchItem = (matchItemBase: MatchListItem, index: number, expanded: string | boolean, handleChangeExpander: (panel: string | boolean) => (event: any, isExpanded: string | boolean) => void) => (
+const returnMatchItem = (
+   {level, matchId, matchNumber, matchStatus, onClickTeam, tableNumber, team1, team2, winnerId}: MatchItemType,
+   index: number,
+   expanded: string | boolean,
+   handleChangeExpander: (panel: string | boolean) => (event: any, isExpanded: string | boolean) => void
+) => (
    <MatchItem
-      level={matchItemBase.level}
-      matchNumber={matchItemBase.matchNumber}
-      matchId={matchItemBase.matchId}
-      matchStatus={matchItemBase.matchStatus}
-      onClickTeam={matchItemBase.onClickTeam}
-      team1={matchItemBase.team1}
-      team2={matchItemBase.team2}
-      winnerTeamId={matchItemBase.winnerId}
-      key={matchItemBase.matchNumber}
+      key={matchNumber}
+      level={level}
+      matchNumber={matchNumber}
+      matchId={matchId}
+      matchStatus={matchStatus}
+      onClickTeam={onClickTeam}
+      tableNumber={tableNumber}
+      winnerTeamId={winnerId}
+      team1={team1}
+      team2={team2}
       expanded={expanded}
       handleChangeExpander={handleChangeExpander}
    />
 )
 
-const returnMatchListItemsFromMatchesListDto = (matchesListDto: MatchesListDto, reloadAllList: () => void): MatchListItem[] => {
-   return matchesListDto.queue.map((matchesItem) => {
-
-      function findStatus(): MatchStatus {
-         if (matchesItem.status === "started") return MatchStatus.STARTED;
-         if (matchesItem.status === "ended") return MatchStatus.FINISHED;
-         if (matchesItem.status === "enqueued") return MatchStatus.NO_TABLE;
-         if (matchesItem.status === "noTeams") {
-            if (matchesItem.team1Id || matchesItem.team1Id) return MatchStatus.NO_ONE_TEAM;
-            return MatchStatus.NO_TEAMS;
+const returnMatchList = (
+   tournamentMatchesListDto: TournamentMatchesListDto,
+   matchesDetailsListDto: MatchDetailsDto[],
+   tournamentTeamsListDto: TournamentTeamsListDto,
+   playersProfilesListDto: PlayerProfileDto[],
+   reloadAllList: () => void
+): MatchItemType[] => {
+
+   const tournamentId = tournamentMatchesListDto.tournamentId;
+   const tournamentMatchesList = {...tournamentMatchesListDto}.queue;
+   const matchesDetailsList = [...matchesDetailsListDto];
+   const tournamentTeamsList = {...tournamentTeamsListDto}.items;
+   const playersProfilesList = [...playersProfilesListDto];
+
+   return tournamentMatchesList
+      .map(
```

Comment:
Warto to co w srodku map dac do innej funkcji - zmniejszysz aagniezdzenie czytajac funkcje - bedzie czytelniej. Tym bardziej przy takiej dlugiej liscie propertiesow z destrukturyzowanego obiektu.

Reply:
hmmm, ok